### ISSUE TITLE
Wrong serial number for HP desktop

### ISSUE THREAD

**Issue by okvok** [fusioninventory/fusioninventory-for-glpi#2508]
We've faced with an issue of wrong serial number. A desktop already in GLPI computer assets with correct serial number SGH327SZYD. Agent scanning the desktop properly as well:
image
![image](https://user-images.githubusercontent.com/20252855/38170012-0e2c3778-3583-11e8-8ba7-a6d5ef599cef.png)

BUT on the server site GLPI 9.2.1 or Fusioninventory plugin 9.2+1.0 change it to GH327SZYD and create a new object in computer assets with a wrong serial number. It loses the first serial number symbol.
image
![image](https://user-images.githubusercontent.com/20252855/38170013-17c6a598-3583-11e8-8e4e-0dcac79bba12.png)

Could anybody help us with this issue? Please!

**Comment by ddurieux** [comment]
It's because the real serial name (on the sticky label) has not the S (the first letter) so we remove it

**Comment by okvok** [comment]
no. It has S on the sticky label. The HP serial number contain 10 symbols.

**Comment by okvok** [comment]
![image](https://user-images.githubusercontent.com/20252855/38171411-e3558b70-35a1-11e8-9991-1e4fe3cad2fc.png)

**Comment by okvok** [comment]
Please fix this BUG.

**Comment by okvok** [comment]
Please help me anybody! Due to the BUG we've stopped the FI plugin!

**Comment by okvok** [comment]
I solved the issue!
Please see below:
![image](https://user-images.githubusercontent.com/20252855/38253935-998ca580-3760-11e8-8011-69d8c83482ff.png)

I think it would be better to exclude this code from the future plugin releases.
Thank you!

**Comment by okvok** [comment]
Yes, we have. But why did you make this patch for HP? I suppose this is wrong patch and it should be excluded from the future releases. I provided you the photo of the sticker with the serial number and you can see S at the beginning of the serial number. May be you will speak to other FI-plugin developers about it?

**Comment by ddurieux** [comment]
Because in old computers the serial number in the BIOS not same than sticker and so have the S un prefix.
Seems they had fix it, but not really simple to manage the old and the new serial number depend on age of the computer

**Comment by AnaelMobilia** [comment]
@ddurieux Really hard to known... Maybe a display as (S)xxxx ? with a popup over the S ?

Regards,
Anael

**Comment by ddurieux** [comment]
Perhaps related with number of characters, must check that (can take some times)
